py4DSTEM/preprocess/preprocess.py

```diff
@@ -630,12 +641,28 @@ def resample_data_diffraction(
 
             resampling_factor = np.array(output_size) / np.array(datacube.shape[-2:])
 
-        resampling_factor = np.concatenate(((1, 1), resampling_factor))
-        datacube.data = zoom(
-            datacube.data, resampling_factor, order=1, mode="grid-wrap", grid_mode=True
-        )
+        output_data = np.zeros(datacube.Rshape + tuple(output_size))
+        for Rx, Ry in tqdmnd(
+            datacube.shape[0],
+            datacube.shape[1],
+            desc="Resampling 4D datacube",
+            unit="DP",
+            unit_scale=True,
+        ):
+            output_data[Rx, Ry] = zoom(
+                datacube.data[Rx, Ry].astype(np.float32),
+                resampling_factor,
+                order=1,
+                mode="grid-wrap",
```

Comment:
shouldn't it be `mode = "nearest"`  ?

Reply:
This came from an early discussion with @sezelt where one of his datasets had severe artifacts during bilinear resampling. Although I don't think we had tried "nearest"?

Reply:
Oh is mode not the edges?  It's the "wrap" part I was worried about here.

Reply:
Changed "wrap" to "nearest" for the DP resampling. There's an equivalent function `process.phase.utils.bilinear_resample` which accepts `grid_mode=True` and `mode="grid-wrap"` flags which default to the values above since we use that function to resample corner-centered exit waves.